### PR TITLE
Issue #89 - fixes industrial_robot_simulator depends

### PR DESCRIPTION
**Pull request by shaun-edwards**
Depends on PR #105

### CODE REVIEW

**Comment by shaun-edwards** [comment]
@gavanderhoorn, could you review.  This addresses your issue #89.  Please verify the dependencies are now correct.  I wasn't quite sure if python dependencies should be `run_depend` only.

**Comment by shaun-edwards** [comment]
@gavanderhoorn, ping/help.

**Comment by gavanderhoorn** [comment]
I don't believe we need all the `build_depend`s in there (at least, not according to [Python module dependencies](http://docs.ros.org/indigo/api/catkin/html/howto/format1/python_module_dependencies.html) from the catkin documentation).

Must say it's becoming increasingly unclear to me what the correct approach is for Python pkgs, but I believe only `run_depend`s are required, unless you are depending on a message pkg.

File: industrial_robot_simulator/package.xml [hunk 1]
```diff
@@ -9,15 +9,12 @@
...
   <build_depend>sensor_msgs</build_depend>
   <build_depend>control_msgs</build_depend>
   <build_depend>trajectory_msgs</build_depend>
-  <build_depend>industrial_robot_client</build_depend>
   <build_depend>industrial_msgs</build_depend>
   <build_depend>python-rospkg</build_depend>
-  <run_depend>roscpp</run_depend>
   <run_depend>std_msgs</run_depend>
   <run_depend>sensor_msgs</run_depend>
   <run_depend>control_msgs</run_depend>
```
Comment (jrgnicho) [inline industrial_robot_simulator/package.xml:20]: Shoulnd't rospy be at least listed as a run_depend?

Comment (gavanderhoorn) [inline industrial_robot_simulator/package.xml:20]: Yes, that should definitely be there.

File: industrial_robot_simulator/CMakeLists.txt [hunk 2]
```diff
@@ -2,22 +2,23 @@ cmake_minimum_required(VERSION 2.8.3)
...
-# ROS launch testing
-## ROS launch testing should be re-enabled when the roslaunch script is officially
-## released (see https://github.com/ros/ros_comm/pull/333)
-##find_package(roslaunch)
-##roslaunch_add_file_check(launch)
+if (CATKIN_ENABLE_TESTING)
+  find_package(roslaunch REQUIRED)
+  roslaunch_add_file_check(launch
+    DEPENDENCIES
+      joint_trajectory_action)
```
Comment (gavanderhoorn) [inline industrial_robot_simulator/CMakeLists.txt:20]: This would be the `industrial_core` one then?

File: industrial_robot_simulator/package.xml (hunk 1 above)
Comment (shaun-edwards) [inline industrial_robot_simulator/package.xml:20]: Added (see below)

**Comment by shaun-edwards** [comment]
Still depends on PR #105.  I think some of the `build_depends` are captured as inherited dependencies and thus not needed.  For now I'd like to leave them in.

File: industrial_robot_simulator/CMakeLists.txt [hunk 3]
```diff
@@ -2,22 +2,23 @@ cmake_minimum_required(VERSION 2.8.3)
...
 ## Test ##
 ##########
 
-# ROS launch testing
-## ROS launch testing should be re-enabled when the roslaunch script is officially
-## released (see https://github.com/ros/ros_comm/pull/333)
-##find_package(roslaunch)
-##roslaunch_add_file_check(launch)
+if (CATKIN_ENABLE_TESTING)
+  find_package(roslaunch REQUIRED)
```
Comment (gavanderhoorn) [inline industrial_robot_simulator/CMakeLists.txt:17]: Adding this will require a `build_depend` on `roslaunch` in the manifest. I suggest:

``` xml
<build_depend version_gte="1.9.55">roslaunch</build_depend>
```

As you're also using the `DEPENDENCIES` arg, which is only supported from that version onwards.

File: industrial_robot_simulator/CMakeLists.txt [hunk 4]
```diff
@@ -2,22 +2,23 @@ cmake_minimum_required(VERSION 2.8.3)
...
+  roslaunch_add_file_check(launch
+    DEPENDENCIES
+      industrial_core)
```
Comment (gavanderhoorn) [inline industrial_robot_simulator/CMakeLists.txt]: `DEPENDENCIES` only takes a list of target names, not package names.

File: industrial_robot_simulator/package.xml [hunk 5]
```diff
@@ -9,15 +9,12 @@
...
 
   <buildtool_depend>catkin</buildtool_depend>
-  <build_depend>roscpp</build_depend>
   <build_depend>std_msgs</build_depend>
   <build_depend>sensor_msgs</build_depend>
   <build_depend>control_msgs</build_depend>
   <build_depend>trajectory_msgs</build_depend>
-  <build_depend>industrial_robot_client</build_depend>
   <build_depend>industrial_msgs</build_depend>
   <build_depend>python-rospkg</build_depend>
```
Comment (gavanderhoorn) [inline industrial_robot_simulator/package.xml]: `python-rospkg` can be deleted. Only `run_depend` needed (see [Python module dependencies](http://docs.ros.org/indigo/api/catkin/html/howto/format1/python_module_dependencies.html)).

File: industrial_robot_simulator/CMakeLists.txt (hunk 4 above)
Comment (shaun-edwards) [inline industrial_robot_simulator/CMakeLists.txt]: `joint_trajectory_action` was the right target then...see [here](https://github.com/ros-industrial/industrial_core/blob/indigo-devel/industrial_robot_client/CMakeLists.txt#L134).  It should be prefixed by the package name, but this is really another issue.

Comment (gavanderhoorn) [inline industrial_robot_simulator/CMakeLists.txt]: Yep, that seems to be the right one.

And, re: prefixing targets: yes, see #45.

Comment (gavanderhoorn) [inline industrial_robot_simulator/CMakeLists.txt]: Although btw, it should be possible to depend on `${industrial_core_EXPORTED_TARGETS}`, iirc.

**Comment by shaun-edwards** [comment]
@gavanderhoorn, can I merge as is?

**Comment by gavanderhoorn** [comment]
No; see in-line comments (args to `DEPENDENCIES` and `python-rospkg` build depend).

**Comment by shaun-edwards** [comment]
OK...failed to push last time...all changes should be in this PR now.

File: industrial_robot_simulator/package.xml [hunk 6]
```diff
@@ -25,4 +22,5 @@
   <run_depend>industrial_robot_client</run_depend>
```
Comment (gavanderhoorn) [inline industrial_robot_simulator/package.xml:22]: One last question: why is this needed? That is a C++ only pkg, is it not?

Comment (shaun-edwards) [inline industrial_robot_simulator/package.xml:22]: There is a launch dependency...[here](https://github.com/ros-industrial/industrial_core/blob/indigo-devel/industrial_robot_simulator/launch/robot_interface_simulator.launch)

Comment (gavanderhoorn) [inline industrial_robot_simulator/package.xml:22]: Yep, forgot about that.

**Comment by gavanderhoorn** [comment]
+1.
